MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -392,4 +405,113 @@ - (void)notifyFailureBeforeSendingForItem:(id<MSLog>)item withError:(NSError *)e
   }
 }
 
-@end
+- (void)stopBackgroundActivity {
+#if !TARGET_OS_OSX
+  if (!MS_IS_APP_EXTENSION) {
+    @synchronized(self) {
```

Comment:
This one won't work as the channel is using a queue for thread management, you need to throw the whole block into the logsDispatchQueue.